template/lambda-sqs-worker-cdk/infra/appStack.ts

```diff
@@ -0,0 +1,65 @@
+import { AccountPrincipal } from '@aws-cdk/aws-iam';
+import { Key } from '@aws-cdk/aws-kms';
+import { AssetCode, Function, Runtime } from '@aws-cdk/aws-lambda';
+import { SqsEventSource } from '@aws-cdk/aws-lambda-event-sources';
+import { Topic } from '@aws-cdk/aws-sns';
+import { SqsSubscription } from '@aws-cdk/aws-sns-subscriptions';
+import { Queue } from '@aws-cdk/aws-sqs';
+import { Construct, Stack, StackProps } from '@aws-cdk/core';
+
+import { envContext, stageContext } from '../shared/context-types';
+
+export class AppStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const stage = stageContext.check(this.node.tryGetContext('stage'));
+    const context = envContext.check(this.node.tryGetContext(stage));
+
+    const accountPrincipal = new AccountPrincipal(this.account);
+
+    const kmsKey = new Key(this, 'kms-key', {
+      description: context.kmsKey.description,
+      enableKeyRotation: true,
+      admins: [accountPrincipal],
+      alias: context.kmsKey.alias,
+    });
+
+    kmsKey.grantEncrypt(accountPrincipal);
+
+    const topic = new Topic(this, 'topic', {
+      topicName: context.topic.name,
```

Comment:
With the CDK, how does one decide between managing configuration values in TypeScript vs context?

It seems to me like the less you have to store in JSON and manually validate the better.

Reply:
Good point re:smaller context. I reckon it's the same concern when it comes to config in `serverless.yaml` versus TypeScript.

For config that's static for any deployment (i.e. to any account), it should be managed in TypeScript. If it needs to vary from account to account, e.g. configuring different reserved concurrency for `dev` versus  `prod` accounts, it should  be in the context.

I'll fix up the config to reflect this.